leases/schema/mutations.py

```diff
@@ -189,6 +189,71 @@ def mutate_and_get_payload(cls, root, info, **input):
         return DeleteBerthLeaseMutation()
 
 
+class SwitchBerthMutation(graphene.ClientIDMutation):
+    class Input:
+        old_lease_id = graphene.ID(required=True)
+        new_berth_id = graphene.ID(required=True)
+        switch_date = graphene.Date(
+            description="The date which will mark the end of the old lease and the start of the new lease."
+            "If none is provided, it will default to the day when the mutation is called"
+        )
+
+    old_berth_lease = graphene.Field(BerthLeaseNode)
+    new_berth_lease = graphene.Field(BerthLeaseNode)
+
+    @classmethod
+    @add_permission_required(BerthLease)
+    @change_permission_required(BerthLease)
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, old_lease_id, new_berth_id, **input):
+        switch_date = input.get("switch_date") or calculate_berth_lease_start_date()
+
+        old_lease = get_node_from_global_id(
+            info, old_lease_id, only_type=BerthLeaseNode, nullable=False,
+        )
+        new_berth = get_node_from_global_id(
+            info, new_berth_id, only_type=BerthNode, nullable=False,
+        )
+
+        end_date = old_lease.end_date
+        try:
+            contract = old_lease.contract
```

Comment:
What if it's a is_non_billable_customer who doesn't have a contract?

Reply:
Good point! We have to address this case as well. So we should just ignore if the contract is missing?

Reply:
Added logic for leases without contracts.